### DISCUSSION TITLE
How to restore the local history that suddenly disappeared ?

### DISCUSSION THREAD

**Discussion by rfrkk** [apache/netbeans#6900] [Q&A]
I started netbeans 18 this morning and found that the local history of all files in the work project suddenly disappeared.
Originally, each file had many history records, but they all disappeared.

ex: All empty
![p1002](https://github.com/apache/netbeans/assets/137764/7652235c-7aea-43ff-a833-91886b2982bd)

However, netbeans places all history files there.
C:\Users\XXX\AppData\Roaming\NetBeans\18\var\filehistory
![p1003](https://github.com/apache/netbeans/assets/137764/f5d53bf9-ae40-4f47-a47f-d49283f0216a)

But, When opening the file, the content becomes garbled.
![p1004](https://github.com/apache/netbeans/assets/137764/a8956db2-b4cd-44c4-bee2-63a0a8bc6ddc)

Please teach me how I can debug and restore the disappeared history.
These contents are very important to me, thank you!

**Comment by junichi11** [comment]
I'm not sure whether we can restore them.
Please try to check your options: Team > Versioning > History
If "Remove Local History files older than (number of days)" is selected, history files are removed.

![nb-options-local-history](https://github.com/apache/netbeans/assets/738383/078cf797-55f0-4dcb-babc-d90bcc0d0ff9)

**Comment by rfrkk** [comment]
@junichi11 
My setup is the same as yours
![p1005](https://github.com/apache/netbeans/assets/137764/b3f36ece-507f-491a-a9a2-d8943f0e0a28)

**Comment by deepvision7** [comment]
That garbled text is a content of a zip archive in fact. You can rename the file to .zip and extract the containing file.
